src/scenario/scenario_events_xml.h

```diff
@@ -0,0 +1,638 @@
+#ifndef SCENARIO_EVENTS_XML_H
+#define SCENARIO_EVENTS_XML_H
+
+#include "building/type.h"
+#include "city/message.h"
+#include "game/resource.h"
+#include "game/settings.h"
+#include "scenario/data.h"
+#include "scenario/scenario_event_data.h"
+
+typedef enum {
+    PARAMETER_TYPE_UNDEFINED = -1,
+    PARAMETER_TYPE_NUMBER = 0,
+    PARAMETER_TYPE_TEXT = 1,
+    PARAMETER_TYPE_CHECK = 2,
+    PARAMETER_TYPE_DIFFICULTY = 3,
+    PARAMETER_TYPE_RESOURCE = 4,
+    PARAMETER_TYPE_BOOLEAN = 5,
+    PARAMETER_TYPE_ROUTE = 6,
+    PARAMETER_TYPE_POP_CLASS = 7,
+    PARAMETER_TYPE_BUILDING = 8,
+    PARAMETER_TYPE_BUILDING_COUNTING = 9,
+    PARAMETER_TYPE_ALLOWED_BUILDING = 10,
+    PARAMETER_TYPE_STANDARD_MESSAGE = 11,
+    PARAMETER_TYPE_FUTURE_CITY = 12,
+    PARAMETER_TYPE_MIN_MAX_NUMBER = 13,
+} parameter_type;
+
+typedef struct {
+    const char *name;
+    parameter_type type;
+    int min_limit;
+    int max_limit;
+} xml_data_attribute_t;
+
+typedef struct {
+    condition_types type;
+    xml_data_attribute_t xml_attr;
+    xml_data_attribute_t xml_parm1;
+    xml_data_attribute_t xml_parm2;
+    xml_data_attribute_t xml_parm3;
+    xml_data_attribute_t xml_parm4;
+    xml_data_attribute_t xml_parm5;
+} scenario_condition_data_t;
+
+static scenario_condition_data_t scenario_condition_data[CONDITION_TYPE_MAX] = {
```

Comment:
These `static` values shouldn't be on a header file, but rather on a C file.

When you need to access, you should return a pointer to their ID via a function on the C file.

Having them as `static` on a header will cause all its values to be duplicated in memory for each different C file that includes this header, which is not what you want.

Reply:
Allright, this was a pretty big refactor.
I moved all these from scenario_events_xml to a new file scenario_events_parameter_data.c
(Not sure on the naming, suggestions welcome.)

Reply:
I imagined it would be, but it was a needed one.

Kudos for adding it to its own file, I think it makes sense that way.

I'll check it out.